Add integration tests for export plugin structure methods

efturtle opened the pull request:
Tests verify exportStructure() methods are properly invoked through Export::exportTable() for plugins that support export structure

Theses tests ensure that when exporting table structure through the Export class, the specific plugin structure export methods are called with the correct parameters

efturtle committed:
Add integration tests for export plugin structure methods

Tests verify exportStructure() methods are properly invoked through Export::exportTable() for plugins that support export structure

Theses tests ensure that when exporting table structure through the Export class, the specific plugin structure export methods are called with the correct parameters

Signed-off-by: efturtle <mexslacker@gmail.com>

efturtle commented:
@kamil-tekiela @williamdes followup on https://github.com/phpmyadmin/phpmyadmin/pull/19996 these are the tests to amplify the test coverage on structure exports.

kamil-tekiela commented on tests/unit/Plugins/Export/ExportHtmlwordTest.php:
```diff
@@ -687,4 +690,66 @@ public function testFormatOneColumnDefinition(): void
+        // Create a mock ExportHtmlword to verify getTableDef is called
+        $relation = new Relation($dbi);
+        $outputHandler = new OutputHandler();
+        $exportHtmlword = $this->getMockBuilder(ExportHtmlword::class)
```
`ExportHtmlword` is the code under test (see line 43) and it **should not** be mocked. So in this case, it makes perfect sense to put a code comment explaining why it's necessary for this test to do something unreasonable. The comment here should explain why `getTableDef` and only that method is mocked.

efturtle replied:
I will change it, thanks for the suggestion it is actually a lot better

<img width="1374" height="908" alt="Image" src="https://github.com/user-attachments/assets/7e5cc489-9199-4e38-8cf2-b931f9b879f3" />

kamil-tekiela reviewed (changes requested):
Thank you so much for these tests. I really appreciate new tests as this makes future work easier. I checked, and they work, but could be improved.

Before this PR is ready to be merged, I have some changes I'd like you to make. Also, I have some general points about this:
- The name `$exportcore` is not a good variable name. For one, it doesn't respect PMA's naming standard, which requires variables to be in camel case. `core` is a term that doesn't appear in the rest of the code or in the class name. 
- Similarly, the test names could be better. `testExportTableStructureThroughExportCore` could be `testExportTableStructureIsCalledThroughExportWhenStructureIsRequested` or `testStructureIsExportedThroughExportClassWhenRequested`. The name should reflect the intent behind the test if it's more than a simple method test. If the tests were in the ExportTest class, it could just be `testStructureIsExportedWhenRequested`. And of course, when the test has a better name, the docblock above it is redundant as it should be.
- As you can see, all of these tests look very similar. But I don't see the positive test for `ExportSql`, which should verify that the settings are still respected. They were moved from Export to ExportSql and we should test that this still works. On the other hand, we only need a single test for all other export plugins that don't have that setting. Perhaps, it would be better to just add two new tests, one in ExportTest and one in ExportSqlTest, one testing that ExportSql still works and the other for an arbitrarily chosen other plugin testing to see if exportStructure is called at all.

efturtle committed:
Refactor export plugin integration tests

- Renamed test methods to testExportTableCallsExportStructureMethod for clarity
- Removed excessive mocking and focused on integration testing
- Simplified test setup by removing unnecessary database/table mocks
- Updated assertions to verify actual export output content
- Made tests more maintainable and easier to understand
- Consistent test structure across ExportHtmlword, ExportLatex, ExportMediawiki, ExportOdt, and ExportPdf plugins

Signed-off-by: Alberto Cuevas <mexslacker@gmail.com>

efturtle commented:
thanks for the suggestions and response on this one. I made a refactor on the tests with your requested changes and recommendations

efturtle committed:
Remove trailing whitespace from ExportPdfTest.php

Signed-off-by: Alberto Cuevas <mexslacker@gmail.com>

kamil-tekiela commented on tests/unit/Plugins/Export/ExportMediawikiTest.php:
```diff
@@ -319,4 +322,47 @@ public function testExportData(): void
+        // Verify that structure output was generated
+        self::assertStringContainsString('Table structure for', $result);
+        self::assertStringContainsString('testtable', $result);
+    }
```
```suggestion
    public function testExportTableCallsExportStructureMethod(): void
    {
        $dbi = $this->getMockBuilder(DatabaseInterface::class)
            ->disableOriginalConstructor()
            ->getMock();

        DatabaseInterface::$instance = $dbi;

        $request = ServerRequestFactory::create()->createServerRequest('POST', 'https://example.com/');
        $this->object->setExportOptions($request, ['mediawiki_structure_or_data' => 'structure']);

        ob_start();
        $export = new Export($dbi, new OutputHandler());
        $export->exportTable(
            'testdb',
            'testtable',
            $this->object,
            null,
            '0',
            '0',
            '',
            [],
        );
        $result = ob_get_clean();

        self::assertIsString($result);
        self::assertStringContainsString('Table structure for', $result);
        self::assertStringContainsString('testtable', $result);
    }
```

kamil-tekiela reviewed (changes requested):
And then please run:
```
composer install
vendor/bin/phpcbf
vendor/bin/psalm --set-baseline=psalm-baseline.xml
vendor/bin/phpstan --generate-baseline
```

efturtle commented on tests/unit/Plugins/Export/ExportPdfTest.php:
```diff
@@ -222,4 +225,55 @@ public function testExportData(): void
+
+        self::assertStringStartsWith('%PDF-', $output);
+        self::assertGreaterThan(5000, strlen($output));
+    }
```
yeah, since it is outputting binary, it is not easy to check with phpunit
best I could check is for line 276 && 277

efturtle committed:
test: export plugin integration tests
- Use this->object instead of creating new plugin instances
- set export options instead of reflection

Signed-off-by: Alberto Cuevas <mexslacker@gmail.com>
